llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -14895,6 +14895,19 @@ static SDValue performSUBCombine(SDNode *N, SelectionDAG &DAG,
   if (SDValue V = combineSubShiftToOrcB(N, DAG, Subtarget))
     return V;
 
+  if (Subtarget.hasStdExtZbb()) {
+    // fold (sub x, (select (ult x, y), 0, y)) -> (umin x, (sub x, y))
```

Comment:
Is there something that prevents this from being

(sub x, (select (uge y, x), y, 0)?

Reply:
That's three differences:
- ult vs uge
- x, y vs y, x
- 0, y vs y, 0

Any two would form a different spelling of the same pattern. I can add more tests. Is there some canonical form or shortcuts to match different spellings?
All three at once (or one) is a different pattern.

Reply:
Oops, I meant `(sub x, (select (uge x, y), y, 0)?` So only inverting the condition and swapping the select order